extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/CustomConfigProducer.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.arc.runtime;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.InjectionPoint;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+
+import io.smallrye.config.inject.ConfigProducerUtil;
+
+/**
+ * This class is the same as io.smallrye.config.inject.ConfigProducer
+ * but uses the proper Quarkus way of obtaining org.eclipse.microprofile.config.Config
+ */
+@ApplicationScoped
+public class CustomConfigProducer {
```

Comment:
Maybe it should be called `QuarkusConfigProducer`?

Reply:
That's a good idea, I'll fix it

Reply:
Done